xarray/core/common.py

```diff
@@ -1152,6 +1152,9 @@ def where(self, cond, other=dtypes.NA, drop: bool = False):
         from .dataarray import DataArray
         from .dataset import Dataset
 
+        if callable(cond):
+            return self.where(cond(self), other=other, drop=drop)
```

Comment:
this is not really a shortcut so the `return` might be misleading (we're calling the method again with different arguments). What about using
```suggestion
            cond = cond(self)
```
instead?

Reply:
Great, that's much cleaner